scripts/Units/armadvsol.bos

```diff
@@ -20,9 +25,51 @@ Stop()
 	return (0);
 }
 
+lua_solarpoint(one, heading) {
+	//get PRINT(1234);
+	//get PRINT(LUA0, LUA1, LUA2, LUA3);//nani
+	return(0);
+}
+
+solarreturn(status, xvector, zvector, sunheading, unitheading) {
+	//get PRINT(1234567);
+	//get PRINT(HEADING);///this seems to always be 82 here? tf?
+	//get PRINT(555,unitheading);
+	//get PRINT(status, xvector, zvector, sunheading);
```

Comment:
so based on DMs this is not forgotten debug code but deliberately left in to figure out an issue:

> [...] not using local HEADING
> instead it goes via lua
> but somehow the heading constant doesn't work in the 'callback' function
> (return values also eaten somehow)
> in theory I shouldn't need to use a callback at all, I should be able to use constants from the returned data
> but it doesn't have them

> basically, the ideal would be this calls lua when done, gets a heading as return value
> no return values possible
> so it has to have a callback function run via lua instead
> and pass the heading along
> as it can't load its own heading in the callback
> for bos reasons

any wizard *bos*ses around? :D

Reply:
see also https://discord.com/channels/549281623154229250/549282804412710925/1219163597012209684

Reply:
In this case using a callback isn't the end of the world, and means that variables are at least nicely named, but it IS bloody weird that HEADING doesn't actually give heading in the callback function, instead I have to refer to the one I passed to lua, then pass back again as a parameter.

Being able to use lua functions with real return values to bos would be useful elsewhere though, so the right magic spell would be appreciated if anyone knows it.

Reply:
print(heading) -> print(get(heading))?

Reply:
Thanks - I've the feeling that that was only the debug incarnation, but old code cleaned a few cycles ago now so who knows. It had a get (heading(0)) originally I'm pretty sure which itself caused issues.